client/admin.go

```diff
@@ -59,31 +65,91 @@ func CreateAndInitTree(ctx context.Context, req *trillian.CreateTreeRequest, adm
 		return nil, err
 	}
 
-	if tree.TreeType == trillian.TreeType_MAP {
-		b.Reset()
-		err := b.Retry(ctx, func() error {
-			glog.Infof("Initialising Map %x...", tree.TreeId)
-			// Now, if it's a Map, initialise it.
-			req := &trillian.InitMapRequest{MapId: tree.TreeId}
-			resp, err := mapClient.InitMap(ctx, req)
-			if err != nil {
-				switch s, ok := status.FromError(err); {
-				case ok && s.Code() == codes.Unavailable:
-					glog.Errorf("Map server unavailable: %v", err)
-					return err
-				case ok && s.Code() == codes.AlreadyExists:
-					glog.Warningf("Bizarrely, the just-created Map (%x) is already initialised!: %v", tree.TreeId, err)
-					return err
-				}
-				return fmt.Errorf("failed to InitMap(%+v): %T %v", req, err, err)
+	switch tree.TreeType {
+	case trillian.TreeType_MAP:
+		err = InitMap(ctx, tree, mapClient)
+	case trillian.TreeType_LOG:
+		err = InitLog(ctx, tree, logClient)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return tree, nil
```

Comment:
Should we guard against unknown tree types here or is doing nothing OK?

Reply:
Added an error for unknown tree types